drivers/ethernet/eth_stm32_hal_priv.h

```diff
@@ -40,7 +40,10 @@ struct eth_stm32_hal_dev_data {
 	struct device *clock;
 	struct k_mutex tx_mutex;
 	struct k_sem rx_int_sem;
-	K_KERNEL_STACK_MEMBER(rx_thread_stack,
+#ifdef CONFIG_SOC_SERIES_STM32H7X
+	struct k_sem tx_int_sem;
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
+	K_THREAD_STACK_MEMBER(rx_thread_stack,
 		CONFIG_ETH_STM32_HAL_RX_THREAD_STACK_SIZE);
 	struct k_thread rx_thread;
```

Comment:
A commit brought new kernel stack definitions. I have resolved the merge conflict during my last rebase. This was tested and working on my Nucleo_H743ZI with HSI 64MHz as clock. PR is still valid.